### PR TITLE
ThreatFixerButton: Add tooltips alt

### PR DESCRIPTION
**Pull request by dkmyta**
## Description
Adds core `Tooltips` instances for the various states of the `ThreatFixerButton` component, as per designs.

Allows the consumer to define the actions performed `onClick`.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

- https://github.com/Automattic/jetpack-scan-team/issues/1524

## Does this pull request change what data or activity we track or use?

- No

## Testing instructions:

* Checkout this branch
* Load storybook with `cd projects/js-packages/storybook && npm run storybook:dev`
* View the Threats Fixer Button and the Threats Data Views component
* Review the various button states and ensure they display and react according to designs
* Review the code and ensure the implementation is correct

## Screenshots
Screen Shot on 2024-11-08 at 13-03-12

Screen Shot on 2024-11-08 at 13-04-14

Screen Shot on 2024-11-08 at 13-04-37

Screen Shot on 2024-11-08 at 13-05-02

Screen Shot on 2024-11-08 at 13-05-25

Screen Shot on 2024-11-08 at 13-05-46

Screen Shot on 2024-11-08 at 13-06-04

Screen Shot on 2024-11-08 at 13-06-24

Screen Shot on 2024-11-08 at 13-06-43

Screen Shot on 2024-11-08 at 13-07-03

Screen Shot on 2024-11-08 at 13-07-21

Screen Shot 2024-11-12 at 10 30 53

Screen Shot on 2024-11-12 at 10-31-29

### CODE REVIEW

File: projects/js-packages/components/components/threat-fixer-button/styles.module.scss [hunk 1]
```diff
@@ -7,3 +7,24 @@
 		box-shadow: none;
 	}
 }
+
+.tooltip {
+	margin-top: 10px;
+	max-width: 240px;
+	border-radius: 4px;
+}
```
Comment (nateweller) [inline projects/js-packages/components/components/threat-fixer-button/styles.module.scss]:
```suggestion
.tooltip {
	margin-top: var( --spacing-base ); // 8px
	max-width: 240px;
	text-align: left;
}
```
Suggestions: use spacing variable, align text to the left for multi-line tooltips, use default border radius.

**Review by nateweller — CHANGES_REQUESTED** [review]
The core tooltip integration looks great! :+1:

The custom error tooltips still only show on hover below the button - what do you think about removing the change here and continuing to use the `ActionPopover` instead of `IconTooltip` for now?

File: projects/js-packages/components/components/threat-fixer-button/styles.module.scss (hunk 1 above)
Comment (dkmyta) [reply projects/js-packages/components/components/threat-fixer-button/styles.module.scss]: Updated to use the suggested in [ee06539](https://github.com/Automattic/jetpack/pull/40111/commits/ee065390c94fdee7e3b55e26c299cd2158f7e20b), but left the `border-radius` to discuss further as the designs currently specify it as `radius-m` which on hover display `4px`:
Screen Shot on 2024-11-12 at 09-58-48

File: projects/js-packages/components/components/threat-fixer-button/index.tsx
Comment (nateweller) [inline projects/js-packages/components/components/threat-fixer-button/index.tsx:1]: Suggestion: with recent changes, `errorMessage` and `tooltipText` are now both exclusively used as the value for the `Tooltip.text` prop. We could compute the value of `tooltipText` as the error message inside the `useMemo` that defines it, and remove the separate declaration of `errorMessage` entirely.

**Review by nateweller — DISMISSED** [review]
LGTM! 😃

**Review by nateweller — APPROVED** [review]
Still looking good! :+1:

**Pull request merged**